lib/internal/repl.js

```diff
@@ -84,17 +99,22 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
     if (data) {
+      repl.history = data.split(/\n|\r/).slice(-repl.historySize);
```

Comment:
`/\n|\r/` will result in empty lines when reading \r\n files. better use `/\r?\n/` or even `/\r?\n|\n?\r/` if you want to be totally safe.